lib/gooddata/models/domain.rb

```diff
@@ -464,47 +452,17 @@ def update_clients_settings(data)
         client_id = datum[:id]
         settings = datum[:settings]
         settings.each do |setting|
-          GoodData::Client.update_setting(setting[:name], setting[:value], domain: self, client_id: client_id)
+          GoodData::Client.update_setting(setting[:name], setting[:value], domain: self, client_id: client_id, data_product_id: datum[:data_product_id])
```

Comment:
Refers to 'datum' more than self (maybe move it to another class?) (FeatureEnvy)